### ISSUE TITLE
[Accessibility] 2xMissing form label, 2xEmpty button, 2xVery low contrast and 1xBroken ARIA reference - Donate pop-up

### ISSUE THREAD

**Issue by cmunteanuRW** [Talleh/VolleyballPub#11]
**Steps to reproduce:**

https://github.com/user-attachments/assets/c7c76e89-f386-4f54-8891-7dddeda5f150

**Actual Result:**

![Image](https://github.com/user-attachments/assets/2a138b1f-22f0-4abc-b073-1fb1771cca38)

**Expected Result:**
No accessibility issues are present
